proto/src/simple.rs

```diff
@@ -226,6 +241,34 @@ impl SimpleBindRequest {
     }
 }
 
+impl CompareRequest {
+    pub fn gen_success(&self) -> LdapMsg {
```

Comment:
Should be gen_compare_true and gen_compare_false since both cases are actually successes, there is no failure. You're pre-generating a true/false response instead.